### PR TITLE
Calendar UI

### PR DESCRIPTION
**Pull request by shinyichen**
There is an issue with displaying nested list in popup description. We are using `dangerouslySetInnerHTML` so we don't really have control there. To resolve that, we may have to change our description to a different format.
Screen Shot 2023-06-09 at 7 53 49 AM

### CODE REVIEW

**Comment by forrestc** [comment]
I converted MD to HTML in buildEvent, maybe I should leave it as MD, and let React parse it by using https://github.com/remarkjs/react-markdown?

File: package.json [hunk 1]
```diff
@@ -5,9 +5,16 @@
   "type": "module",
   "private": true,
   "dependencies": {
+    "@chakra-ui/react": "^2.7.0",
+    "@emotion/react": "^11.11.1",
```
Comment (forrestc) [inline package.json:9]: Are they different fork of react? should the original v18 be taken out?

File: package.json [hunk 2]
```diff
@@ -5,9 +5,16 @@
...
+    "@emotion/styled": "^11.11.0",
+    "@fullcalendar/core": "^6.1.8",
+    "@fullcalendar/daygrid": "^6.1.8",
+    "@fullcalendar/react": "^6.1.8",
```
Comment (forrestc) [inline package.json:13]: another version of react?

File: src/App.js [hunk 3]
```diff
@@ -1,22 +1,43 @@
-import logo from './logo.svg';
-import './App.css';
-import events from './events.js';
+import { Calendar } from "./components/Calendar.jsx";
+import events from "./events.js";
+import { ChakraProvider, Container, Heading, VStack } from "@chakra-ui/react";
+
+// for checking valid date format
+const dateFormat =
+  /^(0[1-9]|1[012])[- /.](0[1-9]|[12][0-9]|3[01])[- /.](19|20)\d\d/;
```
Comment (forrestc) [inline src/App.js]: 👍

**Review by forrestc — APPROVED** [review]
LGTM, thanks @shinyichen! 👍 

You can go ahead and merge it, and I will take care the md-to-react conversion.

File: package.json (hunk 1 above)
Comment (shinyichen) [reply package.json:9]: Chakra is a UI component library, it makes everything look more consistent, like bootstrap but better. `emotion` is required by chakra

File: package.json (hunk 2 above)
Comment (shinyichen) [reply package.json:13]: these are fullcalendar required modules

**Comment by shinyichen** [comment]
Just added activity to the title and groups to the details. What do you think? I think title needs activity, but looks better without. 
Screen Shot 2023-06-09 at 10 45 23 AM

**Comment by forrestc** [comment]
@shinyichen I think it will be looking good if you color different activity types, e.g. blue for retreat, orange for hg, and mark the activity type in the details (not on the title). What do you think?

**Comment by shinyichen** [comment]
Ya makes sense.

**Pull request merged**